portality/static/js/dashboard.js

```diff
@@ -41,157 +67,169 @@ doaj.dashboard.loadGroupTab = function(groupId) {
+            {"index.application_type.exact": "new application"}    // only see open applications, not finished ones
+        ], [{"admin.date_applied": {"order": "asc"}}]);
+
+        return `<li>
+            <span class="label tag tag--featured">Unassigned</span>
+            <a href="${doaj.dashboard.context.applicationsSearchBase}?source=${appUnassignedSource}" class="tag tag--tertiary" title="See unassigned applications">${data.unassigned.applications} <span class="sr-only">applications</span></a>
+        </li>`;
+    }
+
+    // Generate the status progress bar
+    _generateStatusProgressBar = function (data) {
+        let appStatusProgressBar = "";
+        for (let status of doaj.dashboard.statusOrder) {
+            if (data.by_status[status]?.applications > 0) {
+                let appStatusSource = _generateSearchQuerySource([
+                    {"admin.editor_group.exact": data.editor_group.name},
+                    {"admin.application_status.exact": status},
+                    {"index.application_type.exact": "new application"}    // only see open applications, not finished ones
+                ], [{"admin.date_applied": {"order": "asc"}}]);
+
+                appStatusProgressBar += `<li class="progress-bar__bar progress-bar__bar--${status.replace(' ', '-')}" style="width: ${(data.by_status[status].applications / data.total.applications) * 100}%;">
                     <a href="${doaj.dashboard.context.applicationsSearchBase}?source=${appStatusSource}" class="progress-bar__link" title="See ${data.by_status[status].applications} ${status} application(s)">
                         <strong>${data.by_status[status].applications}</strong>
                     </a></li>`;
             }
         }
+        return appStatusProgressBar;
     }
 
-    // ~~-> ApplicationSearch:Page~~
-    let appGroupSource = doaj.searchQuerySource({
-        "term" : [
-            {"admin.editor_group.exact" : data.editor_group.name},
-            {"index.application_type.exact" : "new application"}    // this is required so we only see open applications, not finished ones
-        ],
-        "sort": [{"admin.date_applied": {"order": "asc"}}]
-    });
-    // ~~-> UpdateRequestsSearch:Page ~~
-    // let urGroupSource = doaj.searchQuerySource({ "term" : [
-    //     {"admin.editor_group.exact" : data.editor_group.name},
-    //     {"index.application_type.exact" : "update request"}    // this is required so we only see open applications, not finished ones
-    // ]})
-    let frag = `<div class="tabs__content card">
-        <h3>
-          ${data.editor_group.name}’s open applications
-          <a href="${doaj.dashboard.context.applicationsSearchBase}?source=${appGroupSource}" class="tag tag--secondary" title="See all ${data.editor_group.name}’s open applications ">${data.total.applications}<span class="sr-only"> applications</span></a>
-        </h3>
+     _generateStatisticsFragment = function (data) {
+        let statisticsFrag = "";
```

Comment:
Somewhere in here I think there's an error, because in the testdrive view for admin, the editor's ready applications contains an `undefined` which I assume comes from somewhere in this js.

![Screenshot from 2024-09-24 16-32-31](https://github.com/user-attachments/assets/3749d699-08cd-470e-8999-cf7ddd9b8c8a)

On the same screenshot, I also notice that "Applications by Status" is a heading with no content.  This may be hang-over from the original code, but it's a bit weird to have it; ideally we wouldn't display this header if the application count is zero.

Reply:
Fixed